src/compute/src/render/join/linear_join.rs

```diff
@@ -165,13 +165,14 @@ impl YieldSpec {
 
         let options = s.split(',').map(|o| o.trim());
         for option in options {
-            let parts: Vec<_> = option.split(':').map(|p| p.trim()).collect();
-            match &parts[..] {
-                ["work", amount] => {
+            let mut iter = option.split(':').map(|p| p.trim());
+            let parts : [_; 3]= std::array::from_fn(|_| iter.next());
+            match parts {
```

Comment:
Would `match (iter.next(), iter.next(), iter.next()) { ... }` also work?

Reply:
Yes, but I figured we can inline `parts`, that's even less code :)